File: ncarg2d/man/spps/sflush.m
.TH SFLUSH 3NCARG "March 1993" UNIX "NCAR GRAPHICS"
.na
.nh
.SH NAME
SFLUSH - Flushes polylines, accumulated through calls to the routines
PLOTIF and PLOTIT, from the SPPS polyline buffer shared by those routines;
updates all open workstations; and flushes all system-level I/O buffers.
.SH STATUS
SFLUSH was called FLUSH in previous versions.  The name was changed
to avoid a naming conflict with a common system routine.
.SH SYNOPSIS
CALL SFLUSH
.SH C-BINDING SYNOPSIS
#include <ncarg/ncargC.h>
.sp
void c_sflush() 
.SH EXAMPLES
Use the ncargex command to see the following relevant examples: 
(none)
.SH ACCESS
To use SFLUSH or c_sflush, load the NCAR Graphics libraries ncarg, ncarg_gks,
and ncarg_c, preferably in that order.  
.SH SEE ALSO
Online:
ngpict, frame, plotif, spps, spps_params, ncarg_cbind
.sp
Hardcopy:  
NCAR Graphics Fundamentals, UNIX Version;
User's Guide for NCAR GKS-0A Graphics
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.

File: ncarg2d/src/libncarg/spps/sflush.f
C
C $Id: sflush.f,v 1.8 2008-07-27 00:17:25 haley Exp $
C
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE SFLUSH
C
C SFLUSH flushes the GKS pen-move buffer as well as executing
C an UPDATE WORKSTATION on all open workstations.  In particular,
C for a CGM workstation, this effects a flush of the system level
C I/O buffers.
C
      INTEGER WKID
C
C Check for an uncleared prior error.
C
      IF (ICFELL('SFLUSH - UNCLEARED PRIOR ERROR',1).NE.0) RETURN
C
C Flush the GKS pen move buffer.
C
      CALL PLOTIF (0.,0.,2)
      IF (ICFELL('SFLUSH',2).NE.0) RETURN
C
C Execute UPDATE WORKSTATION on all open workstations.
C
C If no workstations are open, return.
C
      CALL GQOPWK (1,IE,NO,ID)
      IF (IE.NE.0) THEN
        CALL SETER ('SFLUSH - ERROR EXIT FROM GQOPWK',3,1)
        RETURN
      END IF
      IF (NO .EQ. 0) RETURN
C
C Update all workstations.
C
      DO 200 I=1,NO
        CALL GQOPWK (I,IE,NO,WKID)
        IF (IE.NE.0) THEN
          CALL SETER ('SFLUSH - ERROR EXIT FROM GQOPWK',4,1)
          RETURN
        END IF
        CALL GUWK(WKID,0)
  200 CONTINUE
C
C Done.
C
      RETURN
C
      END
